language-plutus-core/src/Language/PlutusCore/TypeSynthesis.hs

```diff
@@ -66,7 +66,7 @@ intop = do
 
 unit :: MonadState Int m => m (Type TyNameWithKind ())
 unit =
-    [ TyForall () nam (Type ()) (TyVar () nam) | nam <- newTyName (Type ()) ]
+    [ TyForall () nam (Type ()) (TyFun () (TyVar () nam) (TyVar () nam)) | nam <- newTyName (Type ()) ]
```

Comment:
It would be nice if the typechecker used the definitions of things that we have in the stdlib. To do that it would need to run in `Quote`, which is probably good anyway, since that provides facilities for making new names. I could put up a PR with that if you'd like?

Reply:
That sounds good to me.